includes/User.php

```diff
@@ -4841,7 +4841,29 @@ protected function saveOptions() {
 			}
 		}
 
-		$dbw->delete( 'user_properties', array( 'up_user' => $this->getId() ), __METHOD__ );
+		// kinda ghetto, but :(
+		if ($wgPreferencesUseService) {
+			$regex = implode ('|', $wgUserPreferenceWhiteList['regexes']);
```

Comment:
I don't think that `User.php` needs to know about the whitelist. Since `PreferencePersistenceMySQL` knows about the whitelist and filters based on that

```php
			$preferenceNames = array_keys($this->userPreferences()->getPreferences($this->getId()));

			(new WikiaSQL())
				->DELETE('user_properties')
				->WHERE('up_user')->EQUAL_TO($this->getId())
					->AND_('up_property')->NOT_IN($preferenceNames)
				->run($dbw);

			$insert_rows = array_reduce($insert_rows, function($result, $current) use ($preferenceNames) {
				if (!in_array($current['up_property'], $preferenceNames)) {
					$result[] = $current;
				}

				return $result;
			}, []);
```

Should produce only whitelisted preference names